common-api/src/main/scala/pl/touk/nussknacker/engine/graph/expression/Expression.scala

```diff
@@ -1,17 +1,32 @@
 package pl.touk.nussknacker.engine.graph.expression
 
 import io.circe.generic.JsonCodec
+import io.circe.syntax.EncoderOps
 
+// TODO in the future 'expression' should be a dedicated type rather than raw string, it would for example make DictLabelWithKeyExpression handling prettier
 @JsonCodec case class Expression(language: String, expression: String)
 
 object Expression {
 
   object Language {
-    val Spel         = "spel"
-    val SpelTemplate = "spelTemplate"
+    val Spel             = "spel"
+    val SpelTemplate     = "spelTemplate"
+    val Literal          = "literal"
+    val DictLabelWithKey = "dictLabelWithKey"
   }
 
   def spel(expression: String): Expression = Expression(Language.Spel, expression)
 
   def spelTemplate(expression: String): Expression = Expression(Language.SpelTemplate, expression)
+
+  def literal(expression: String): Expression = Expression(Language.Literal, expression)
+
+  @JsonCodec
+  case class DictLabelWithKeyExpression(dictId: String, label: String, key: String)
```

Comment:
1. Why we need `dictId`? It is already available in the definition of component (in the editor). Which dictId will be more important - this stored in the json or in the definition?
2. I can't see any logic in the scenario UI resolving part. How to you handle changes in the label?

Reply:
i've fiddled with this for a while, and it's hard to do nicely because of `expression: String`, i've added a TODO to change it, but I can't do it right now

I think the way I did it (one class with optional label) is slightly better because of only one parser